jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/AnsiLoggerWithFooter.java

```diff
@@ -93,49 +114,57 @@ public void awaitTermination() {
    *
    * <p>The footer is printed in <strong>bold</strong>.
    *
-   * @param footerLines the footer, with each line as an element (no newline at end)
+   * @param newFooterLines the footer, with each line as an element (no newline at end)
    * @return a {@link Future} to track completion
    */
-  public Future<Void> setFooter(List<String> footerLines) {
-    if (footerLines.equals(this.footerLines)) {
+  public Future<Void> setFooter(List<String> newFooterLines) {
+    if (newFooterLines.equals(footerLines)) {
       return Futures.immediateFuture(null);
     }
 
-    return log(() -> {}, footerLines);
-  }
-
-  private Future<Void> log(Runnable messageLogger, List<String> newFooterLines) {
     return executorService.submit(
         () -> {
-          StringBuilder plainLogBuilder = new StringBuilder();
+          boolean didErase = eraseFooter();
 
-          // Moves the cursor up to the start of the footer.
-          // TODO: Optimize to single init.
-          for (int i = 0; i < this.footerLines.size(); i++) {
-            // Moves cursor up.
-            plainLogBuilder.append(CURSOR_UP_SEQUENCE);
-          }
+          // If a previous footer was erased, the first new footer line needs to go up a line.
+          String newFooterPrefix = didErase ? CURSOR_UP_SEQUENCE : "";
 
-          // Erases everything below cursor.
-          plainLogBuilder.append(ERASE_DISPLAY_BELOW);
-
-          // Writes out logMessage and footer.
-          plainLogger.accept(plainLogBuilder.toString());
-          messageLogger.run();
-
-          if (newFooterLines.size() > 0) {
-            StringBuilder footerBuilder = new StringBuilder();
-            for (String newFooterLine : newFooterLines) {
-              footerBuilder.append(BOLD).append(newFooterLine).append(UNBOLD).append('\n');
-            }
-            // Removes last newline.
-            footerBuilder.setLength(footerBuilder.length() - 1);
-            plainLogger.accept(footerBuilder.toString());
+          for (String newFooterLine : newFooterLines) {
+            plainPrinter.accept(newFooterPrefix + BOLD + newFooterLine + UNBOLD);
+            newFooterPrefix = "";
```

Comment:
Hmm... how unfortunate. What do you think of

```java
plainPrinter.accept(TWO_TIMES_CURSOR_UP_SEQUENCE)?
for (...) {
  plainPrinter.accept(BOLD + newFooterLine + UNBOLD);
}
```
?

Reply:
Yea that would be ideal, though somehow on Gradle, this sometimes causes a prior-logged line to disappear. Not too sure why that happens though...

Reply:
Ah, OK. Moving one line seems safter.